### ISSUE TITLE
0.4.3 breaks cache behavior for plugins which change the file extension

### ISSUE THREAD

**Issue by janb87** [jgable/gulp-cache#56]
I'm using gulp cache in combination with gulp-less.

After the latest release (0.4.3) my build broke.

What happens if my file is in the cache:
0.4.2 Build styles.less => outputs styles.css
0.4.3 Build styles.less => outputs styles.less

```javascript
var gulp = require('gulp');
var less = require('gulp-less');
var cache = require('gulp-cache');

gulp.src('**/*.less')
                .pipe(cache(less(), {
                    key: function (file) {
                        return [file.contents.toString('utf8'), file.path, 'less'].join('');
                    },
                    success: function (file) {
                        console.log('Compiled ' + file.path);
                        return true;
                    }
                }))
                .pipe(gulp.dest('.'));
```

**Comment by jgable** [comment]
That's unfortunate.  I think you'll need to add the gulp-less version to the key to be extra safe for this kind of thing.  I've done this manually or by requiring the package.json and doing it programmatically.

We can't predict all the factors that will cause the cache to no longer be valid; we can make some good faith guesses and common sense defaults at best.

**Comment by janb87** [comment]
@jgable This will not solve the issue.
The problem is that the last change which allows you to cache files with different names but the same content changed behavior. 

Create a PR for this, please have a look.

**Comment by jgable** [comment]
Fixed by #57 and published as 0.4.4
